### DISCUSSION TITLE
What is the range of seq_id?

### DISCUSSION THREAD

**Discussion by raulod** [ggml-org/llama.cpp#9885] [Q&A]
can `llama_seq_id` random or arbitrary large value in `llama_batch_add` or other functions? will it affect performance?

**Comment by slaren** [comment]
You have to set the maximum sequence id in `llama_context_params::n_seq_max` when creating the context. It shouldn't affect performance, but it may affect memory usage of recurrent models.

**Reply by raulod** [reply]
thank you, is this because kv cache maintained per seq_id?

**Reply by slaren** [reply]
In transformers with standard attention the value `n_seq_max` does not make a difference for the KV cache. In recurrent models I believe that there is one context stored for each sequence.
